_posts/2020-02-24-qute.adoc

```diff
@@ -0,0 +1,198 @@
+<2> `{title}` is not validated - not matching a param declaration.
+<3> `{foo.message}` is validated. `org.acme.Foo` must have a property `message` or a matching template extension method must exist. 
+<4> `{foo.message}` is not validated because `foo` is overridden in the loop section and there is no type information available.
+
+=== First-class Quarkus Citizen
+
+In Quarkus, all templates located in the `src/main/resources/templates` directory are validated and can be easily injected.
+
+.Template Injection Example
+[source,java]
+----
+package org.acme.qute;
+
+import io.quarkus.qute.Template;
+
+class MyBean {
+
+    @Inject 
+    Template items; <1>
+    
+    @Inject
+    Service service;
+
+    String renderItems() {
+       return items.data("items", service.getItems()).render(); <2>
+    }
+}
+----
+<1> The field name is used to locate the template. In this particular case, the container will attempt to locate a template with path `src/main/resources/templates/items.html`. If there is no such template available the build fails.
+<2> See the https://quarkus.io/guides/qute-reference#hello-world-example[Hello World Example] to explore the basic workflow.
+
+Moreover, a preconfigured `Engine` instance is provided and available for injection.
+The `Engine` is a central point for template management and provides some low-level API.
+
+=== RESTEasy Integration
+
+If used together with RESTEasy a resource method may return a `TemplateInstance` and the integration code takes care of all the necessary steps and renders the output to the response. 
+See https://quarkus.io/guides/qute-reference#resteasy-integration[RESTEasy Integration] for more information.
+
+.JAX-RS Resource Example
+[source,java]
+----
+package org.acme.qute;
+...
+import io.quarkus.qute.TemplateInstance;
+import io.quarkus.qute.Template;
+
+@Path("hello")
+public class HelloResource {
+
+    @Inject
+    Template hello; <1>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public TemplateInstance get(@QueryParam("name") String name) {
+        // the template looks like: Hello {name}!
+        return hello.data("name", name); <2> <3>
+    }
+}
+----
+<1> The field name is used to locate the template. In this particular case, we're injecting a template with path `templates/hello.txt`.
+<2> `Template.data()` returns a new template instance that can be customized before the actual rendering is triggered. In this case, we put the name value under the key `name`. The data map is accessible during rendering. 
+<3> Note that we don't trigger the rendering - this is done automatically by a special `ContainerResponseFilter` implementation.
+
+
+=== Mailer Integration
+
+Templates may come in handy when sending e-mails.
+The Mailer extension integrates with Qute to provide a convenient way of sendig e-mails.
+In particular, the message body is automatically created using `\*.html` and `*.txt` templates from the `src/main/resources/templates` directory. 
```

Comment:
```suggestion
In particular, the message body is automatically created using `*.html` and `*.txt` templates from the `src/main/resources/templates` directory. 
```

Reply:
Hm, without the baskslash it's not rendered properly in my editor. I should probably use the inline literal syntax...